### PR TITLE
[Meghana-step 10,12,13] Search and Sort in the task list 

### PR DESCRIPTION
**Pull request by Meghana-Reddy2**
Done 

- Added the search feature to able to search by name , description , priority,status,duedate
- Added sort feature to name ,description , priority ,status,duedate 
-Added the system spec 

Screenshot 2024-02-29 at 17 42 35
Screenshot 2024-02-29 at 17 42 48
Screenshot 2024-02-29 at 17 35 41

### CODE REVIEW

File: myapp/Gemfile [hunk 1]
```diff
@@ -60,3 +60,4 @@ end
 gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 gem 'rails-i18n'
+gem 'ransack'
```
Comment (hayashiR1) [inline myapp/Gemfile]: Please add new line 🙏 
(Rails needs a empty line at the end of file)

File: myapp/spec/system/tasks_spec.rb [hunk 2]
```diff
@@ -74,10 +74,66 @@
...
+
+      it 'by duedate asc' do
+        visit '/'
+        click_link('Duedate')
+        expect(page.body.index('Task1')).to be < page.body.index('Task3')
+        expect(page.body.index('Task3')).to be < page.body.index('Task2')
+        click_link('Duedate')
+        expect(page.body.index('Task2')).to be < page.body.index('Task3')
+        expect(page.body.index('Task3')).to be < page.body.index('Task1')
+      end
```
Comment (hayashiR1) [inline myapp/spec/system/tasks_spec.rb]: Please remove indent 🙏 
```suggestion
    it 'by duedate asc' do
      visit '/'
      click_link('Duedate')
      expect(page.body.index('Task1')).to be < page.body.index('Task3')
      expect(page.body.index('Task3')).to be < page.body.index('Task2')
      click_link('Duedate')
      expect(page.body.index('Task2')).to be < page.body.index('Task3')
      expect(page.body.index('Task3')).to be < page.body.index('Task1')
    end
```

File: myapp/app/models/task.rb [hunk 3]
```diff
@@ -4,4 +4,8 @@ class Task < ApplicationRecord
     validates :name, presence: true
     validates :status, presence: true, inclusion: { in: STATUSES }
     validates :priority, presence: true, inclusion: { in: PRIORITY }
+
+    def self.ransackable_attributes(auth_object= nil)
+        ["name", "description", "status", "priority", "duedate"]
+    end
```
Comment (hayashiR1) [inline myapp/app/models/task.rb]: Indent of code guideline in Rakuma is two spaces.
so please remove spaces 🙏 

```suggestion
    def self.ransackable_attributes(auth_object= nil)
      ["name", "description", "status", "priority", "duedate"]
    end    
```

File: myapp/app/views/tasks/index.html.erb [hunk 4]
```diff
@@ -21,16 +27,28 @@
         <td><%= task.priority %></td>
         <td><%= task.status %></td>
         <td><%= task.duedate %></td>
+        <td><%=  %></td>
```
Comment (hayashiR1) [inline myapp/app/views/tasks/index.html.erb]: is this necessary? 😅

**Review by hayashiR1 — COMMENTED** [review]
I posted some comments, please check it 🥺

File: myapp/Gemfile (hunk 1 above)
Comment (Meghana-Reddy2) [reply myapp/Gemfile]: I will remember this 😅

File: myapp/spec/system/tasks_spec.rb (hunk 2 above)
Comment (Meghana-Reddy2) [reply myapp/spec/system/tasks_spec.rb]: Done.

File: myapp/app/models/task.rb (hunk 3 above)
Comment (Meghana-Reddy2) [reply myapp/app/models/task.rb]: Done.

File: myapp/app/views/tasks/index.html.erb (hunk 4 above)
Comment (Meghana-Reddy2) [reply myapp/app/views/tasks/index.html.erb]: I didn't want my 'Edit' button to be displayed under the 'created_at' column so added that line 😅 . But I changed it to  <td><%= "\t" %></td>. Please check it once 🙏

Comment (hayashiR1) [reply myapp/app/views/tasks/index.html.erb]: sorry i didn't understand correctly, but you're right, you need empty td tag this line!

**Review by hayashiR1 — APPROVED** [review]
264203072-ba6e3ed0-17a5-4e0b-882d-00dbe4dc5b8f

**Pull request merged**